### ISSUE TITLE
Zulip Integration

### ISSUE THREAD

**Issue by iammattmartin** [chatwoot/chatwoot#2619]
**Describe the solution you'd like**
Whilst there is native Slack, this assumes that everyone uses that and keeps their data in the cloud. It would be nice to use some options where cloud and on-prem is supported.

Zulip remains another option and has a very good API for messaging (https://zulip.com/api/send-message) but also two-way changes within the app (https://zulip.com/api/get-events).

Given Zulips streams/topics, a new topic could be created for each visitor allowing segregation of conversations. 

We've switched recently from Rocket.Chat to Zulip and find the workflow much better.

Even initially just one-way notification (webhook etc) would be great but two-way to answer, resolve and so on would be even better.

**Comment by wokawoka** [comment]
since Zulip should be slack API compatible (https://zulip.com/integrations/doc/slack , https://zulip.com/integrations/doc/slack_incoming ), would it be possible to integrate it with Chatwoot through the Slack integration?

**Comment by iammattmartin** [comment]
Nope, the slack integration is more app based than just webhook calls.
